databricks/koalas/series.py

```diff
@@ -4701,16 +4703,43 @@ def repeat(self, repeats: int) -> "Series":
         >>> ks.Series([1, 2, 3]).repeat(0)
         Series([], Name: 0, dtype: int64)
         """
-        if not isinstance(repeats, int):
-            raise ValueError("`repeats` argument must be integer, but got {}".format(type(repeats)))
-        elif repeats < 0:
-            raise ValueError("negative dimensions are not allowed")
+        if not isinstance(repeats, (int, Series)):
+            raise ValueError(
+                "`repeats` argument must be integer or Series, but got {}".format(type(repeats))
+            )
 
-        kdf = self.to_frame()
-        if repeats == 0:
-            return first_series(DataFrame(kdf._internal.with_filter(F.lit(False))))
+        if isinstance(repeats, Series):
+            if LooseVersion(pyspark.__version__) < LooseVersion("2.4"):
```

Comment:
@ueshin, I think we officially support Spark 2.4+ so maybe we wouldn't have to worry about this case.

Reply:
It's okay because it's just a better error message though :-)

Reply:
Yeah, maybe we can remove it after we make Koalas throw an error instead of warning (https://github.com/databricks/koalas/blob/master/databricks/koalas/__init__.py#L35-L40) when using Spark<2.4.